jsk_perception/src/remove_blurred_frames.cpp

```diff
@@ -0,0 +1,90 @@
+ *     contributors may be used to endorse or promote products derived
+ *     from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+ *  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+ *  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+ *  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+ *  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ *  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+ *  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+ *  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ *  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+ *  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ *  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+/*
+ * remove_blurred_frames.cpp
+ * Author: Yoshiki Obinata <obinata@jsk.imi.i.u-tokyo.ac.jp>
+ */
+
+#include "jsk_perception/remove_blurred_frames.h"
+
+namespace jsk_perception{
+    void RemoveBlurredFrames::onInit(){
+        DiagnosticNodelet::onInit();
+        srv_ = boost::make_shared<dynamic_reconfigure::Server<Config> > (*pnh_);
+        dynamic_reconfigure::Server<Config>::CallbackType f =
+            boost::bind(&RemoveBlurredFrames::configCallback, this, _1, _2);
+        srv_->setCallback (f);
+        pub_ = advertise<sensor_msgs::Image>(*pnh_, "output", 1);
+        pub_masked_ = advertise<sensor_msgs::Image>(*pnh_, "output/mask", 1);
+        onInitPostProcess();
+    }
+
+    void RemoveBlurredFrames::subscribe(){
+        sub_ = pnh_->subscribe("input", 1, &RemoveBlurredFrames::work, this);
+    }
+
+    void RemoveBlurredFrames::unsubscribe(){
+        sub_.shutdown();
+    }
+
+    void RemoveBlurredFrames::work(const sensor_msgs::Image::ConstPtr& image_msg){
+        cv::Mat image, gray, laplacian_image, masked_image;
+        sensor_msgs::Image::Ptr mask_img_msg;
+        cv::Scalar mean, stddev;
+        double var;
+        vital_checker_ -> poke();
+        boost::mutex::scoped_lock lock(mutex_);
+        image = cv_bridge::toCvShare(image_msg, image_msg->encoding) -> image;
+        cv::cvtColor(image, gray, cv::COLOR_RGB2GRAY);
+        cv::Laplacian(gray, laplacian_image, CV_64F);
+        cv::meanStdDev(laplacian_image, mean, stddev, cv::Mat());
+        var = stddev.val[0] * stddev.val[0];
+        NODELET_DEBUG("%s : Checking variance of the Laplacian: actual %f, threshold %f\n", __func__, var, min_laplacian_var_);
+        laplacian_image.convertTo(masked_image, CV_8UC1, 255.0);
+        mask_img_msg = cv_bridge::CvImage(image_msg->header, sensor_msgs::image_encodings::MONO8, masked_image).toImageMsg();
+        pub_masked_.publish(mask_img_msg);
+        if(var > min_laplacian_var_){
```

Comment:
It may be that the user wants to use this value, so it is better to publish score (`var`).

Reply:
updated